app/services/vot/parser.rb

```diff
@@ -0,0 +1,96 @@
+module Vot
+  class Parser
+    class ParseException < StandardError; end
+    Result = Data.define(
+      :component_values,
+      :aal2?,
+      :phishing_resistant?,
+      :hspd12?,
+      :identity_proofing?,
+      :biometric_comparison?,
+      :ialmax?,
+    )
+
+    attr_reader :vector_of_trust
+
+    def initialize(vector_of_trust)
+      @vector_of_trust = vector_of_trust
+    end
+
+    def parse
+      initial_components = map_initial_vector_of_trust_components_to_component_values
+      expand_components_with_initial_components(initial_components)
+    end
+
+    def parse_acr
+      initial_components = map_initial_acr_values_to_component_values
+      expand_components_with_initial_components(initial_components)
+    end
+
+    private
+
+    def map_initial_vector_of_trust_components_to_component_values
+      vector_of_trust.split('.').map do |component_value_name|
+        component_value = SupportedComponentValues.by_name[component_value_name]
+        if component_value.nil?
+          raise_unsupported_component_exception(component_value_name)
+        end
+        component_value
+      end
+    end
+
+    def map_initial_acr_values_to_component_values
+      vector_of_trust.split(' ').map do |component_value_name|
+        component_value = LegacyComponentValues.by_name[component_value_name]
+        if component_value.nil?
+          raise_unsupported_component_exception(component_value_name)
+        end
+        component_value
+      end
```

Comment:
If we switch to `Hash#fetch` we can get a KeyError and just rescue that (or even let it raise if we want to)

```suggestion
        LegacyComponentValues.by_name.fetch(component_value_name)
      rescue KeyError
        raise_unsupported_component_exception(component_value_name)
      end
```